### ISSUE TITLE
Potential vulnerability stopping the P2P network altogether rather quickly

### ISSUE THREAD

**Issue by dumblob** [threefoldtech/zos#1640]
The following is based on my (very) shallow understanding of ThreeFold. So bare with me.

It seems the bootstrap process depends on contacting a hardcoded list of existing "Gateway" servers in the P2P network (or worse some hardcoded public servers which are not even part of the P2P network). It doesn't seem to use any means to circumvent even the most basic ways of filtering and blocking (e.g. port blocking, DNS name blocking, ...) not even speaking about DPI /deep packet inspection/ (I don't know of any state nor big telco provider not having it on backbones!), etc. At least encrypted DHT over random ports would be nice as the initial step (see e.g. https://medium.com/unitychain/intro-to-dht-e98425fc05f1 and https://medium.com/unitychain/dht-attacks-and-defenses-e159b3d1bcf8 ).

What happens with the ThreeFold network if somehow these very few designated servers become unavailable (e.g. due to a DDoS or data center outage or DNS blocking or whatever) for more than a very short period during which the peers in the network will be forced to stop doing their job due to some timeouts or even perform a reboot (maybe by another attack - MITM or whatever).

I suppose the **whole network** will immediately stop working. And of course there won't be any way to resurrect the network because even if new nodes would want to join, they wouldn't have anyone to connect to because of all the few `rendezvous` servers unavailable.

Do you have any documentation about attack vectors on ThreeFold and how does the security profile look like?

**Comment by scottyeager** [comment]
Yes, there are indeed currently some centralized aspects that represent single points of failure. I also wrote about DNS dependency for accessing public TF Chain nodes in #1573.

These are roles that I imagine our L0 validators could take on: hosting TF Chain nodes and also hosting the Zos code, with some kind of robust peer discovery system.

We are thinking right now about how to make the system "unbreakable", but I don't believe any comprehensive docs exist on the current state and risks.

**Comment by coesensbert** [comment]
Sorry about a late reply on this.
We are very aware of the current issues around mainnet endpoints and ways of communication / interaction with grid services. I think it's not a good idea to explain (in public) in details what these issues are, but know that we already put a lot of effort in figuring out and testing a better setup. We are building a combination of HA and GEO capable layered backend services which will make every endpoint more secure, less (of almost no) chance of going down and will route you to the closest available endpoint. All of this is build with scaling in mind, so we adhere the quest for decentralization.
We will also implement extensive monitoring to see how the backend performs, what can be improved and how it behaves when there are network disruptions.
Things will improve! But we need some time to build, test and implement. We are of course always open for discussion or suggestions.

**Comment by dumblob** [comment]
Thanks for clarification. Feel free to contact me over email at {mygithubnick}at{gmail}dot{com} to keep me informed about the progress on this front (I might even contribute some thoughts if there is any interest). It's unfortunately a showstopper for the use cases I have in mind for ThreeFold.
